woocommerce/woocommerce-ios
# [Order List Redesign] Add starter view for Order Search

Refs #956. 

This is the second PR for implementing the list of Order Statuses shown in Orders → Search. The first part is https://github.com/woocommerce/woocommerce-ios/pull/1804 and should be reviewed first. 

Behavior | Large Font
--------|-------
<img src="https://user-images.githubusercontent.com/198826/73494057-1e7b8b80-4371-11ea-841c-e831c499b0a3.gif" width="320">| <img src="https://user-images.githubusercontent.com/198826/73494147-3e12b400-4371-11ea-8134-245c76ae54d0.png" width="320">

Light | Dark
--------|-------
![image](https://user-images.githubusercontent.com/198826/73494177-4c60d000-4371-11ea-835c-4ea623d9c3c4.png)        |       ![image](https://user-images.githubusercontent.com/198826/73494182-4ec32a00-4371-11ea-9b09-58c46dffdff0.png)

## Changes

This mainly adds the implementation of `OrderSearchUICommand.createStarterViewController()`. That protocol method was added in #1804.  

https://github.com/woocommerce/woocommerce-ios/blob/3af0a89433c31336287a86325eb032577cb34553/WooCommerce/Classes/ViewRelated/Search/Order/OrderSearchUICommand.swift#L33-L35

For `OrderSearchUICommand`, the new class named `OrderSearchStarterViewController` is returned. This `ViewController` queries and shows the order statuses of the current site.

### Dev Builds Only

This change is hidden behind a feature flag and is not visible in production builds. For production builds, the previous search implementation is left intact.

## What's Next

Tapping on the statuses currently has no effect. This will be implemented next.

## Reviewing 

1. Review #1804 first. 
2. Then review this. 🙂 

## Testing 

1. Navigate to Orders → Search
2. Confirm that the Order Statuses view is shown when there is no keyword entered.
3. Confirm that the Order Statuses view is hidden when there are search results. 
4. Confirm that if the feature flag is off, the previous search implementation is displayed.

## Submitter Checklist
